.travis.yml

```diff
@@ -36,9 +36,17 @@ jobs:
     - if: type = pull_request
       name: checkpatch
       script:
+        - *short_log
+        # TODO: stop screen-scraping the git log and switch to
+        # ${TRAVIS_COMMIT_RANGE} or ${TRAVIS_BRANCH}..HEAD after testing
+        # both force pushes and not.
+        # Note TRAVIS_COMMIT_RANGE has triple dots for "git diff" and
+        # needs conversion to double dots for git log's inconsistent
+        # user interface.
+        - printf '%s %s\n' "${TRAVIS_BRANCH}" "${TRAVIS_COMMIT_RANGE/.../..}"
         - SHA_PR=`git log --oneline -1 | sed -rn "s/.*Merge (.*) into.*/\1/p"`
         - SHA_MAIN=`git log --oneline -1 | sed -rn "s/.*Merge .* into (.*)/\1/p"`
-        - scripts/checkpatch.pl --strict --codespell -g $SHA_MAIN..$SHA_PR
+        - ( set -x; scripts/checkpatch.pl --strict --codespell -g $SHA_MAIN..$SHA_PR )
```

Comment:
I don't know what set -x does to be honest?

Reply:
When you look at https://travis-ci.org/github/thesofproject/linux/jobs/711539380 you see that Travis prints:
```shell
scripts/checkpatch.pl --strict --codespell -g $SHA_MAIN..$SHA_PR
```
thanks to `set -x` it also prints:
```shell
++scripts/checkpatch.pl --strict --codespell -g 85b281e277a..68f0916064
```
This is useful because SHA_MAIN and SHA_PR are "screen-scrapped" from the git loghttps://en.wikipedia.org/wiki/Data_scraping. 

PS: `set -x` or `[ba]sh -x foo.sh` tracing is the very first action to take when something goes wrong with a script. Travis has a complex shell execution wrapper to have fine control over how it log commands. Other CI engines don't bother and just `set -x` everything. Comparing a CI trace with a local trace is often the only way to find unwanted CI differences.